Support Java 11 (7.x)

zoliMolnar opened the pull request:
This pull request adds minimalistic support for Java 11.

Commit:
[upgrade] Target platform changes for Java 11

Commit:
[upgrade] Compile error fixes, spring upgrade to 5.1.1.RELEASE

Commit:
[upgrade] Manifest changes

Commit:
[upgrade] Classpath changes

Commit:
[upgrade] Target platform changes for the 4.10-rc2 eclipse build

Commit:
Merge '7.x' branch into 'feature/java_11' branch

Commit:
[upgrade] Add javax maven dependencies

Commit:
[upgrade] Add javax.xml.bind as an imported package

Commit:
Merge remote-tracking branch 'origin/7.x' into feature/java_11

Conflicts:
	CHANGELOG.md
	commons/com.b2international.collections.fastutil/.classpath
	commons/com.b2international.commons/.classpath
	commons/com.b2international.commons/META-INF/MANIFEST.MF
	commons/com.b2international.index/.classpath
	commons/com.b2international.index/src/com/b2international/index/es/EsDocumentSearcher.java
	commons/com.b2international.mapdb/.classpath
	core/com.b2international.snowowl.datastore/src/com/b2international/snowowl/datastore/request/version/CodeSystemVersionCreateRequest.java
	core/com.b2international.snowowl.fhir.api.tests/src/com/b2international/snowowl/fhir/tests/SandBoxRestTest.java
	core/com.b2international.snowowl.identity.file/.classpath
	core/com.b2international.snowowl.identity.file/META-INF/MANIFEST.MF
	dependencies/com.b2international.snowowl.hibernate.validator/.classpath
	dependencies/com.b2international.snowowl.hibernate.validator/META-INF/MANIFEST.MF
	dependencies/org.supercsv/.classpath
	dependencies/org.supercsv/META-INF/MANIFEST.MF
	pom.xml
	releng/target-platform/.launch/snow-owl-oss.launch
	snomed/com.b2international.snowowl.snomed.api.rest.tests/src/com/b2international/snowowl/snomed/api/rest/branches/SnomedMergeApiTest.java
	snomed/com.b2international.snowowl.snomed.datastore.tests/META-INF/MANIFEST.MF
	snomed/com.b2international.snowowl.snomed.datastore/src/com/b2international/snowowl/snomed/datastore/request/SnomedRefSetMemberSearchRequest.java
	tests/com.b2international.restassured-all/.classpath
	tests/com.b2international.restassured-all/META-INF/MANIFEST.MF

Commit:
Merge remote-tracking branch 'origin/7.x' into feature/java_11

Commit:
[upgrade] Remove inner class and reflection

apeteri commented on releng/target-platform/.launch/snow-owl-oss.launch:
```diff
@@ -22,8 +22,13 @@
 <stringAttribute key="pde.version" value="3.3"/>
 <stringAttribute key="product" value="com.b2international.snowowl.server.product.product"/>
 <stringAttribute key="productFile" value="\com.b2international.snowowl.server.update\so_server.product"/>
+<<<<<<< HEAD
```
⚠️ Merge conflict markers are still present in this file.

apeteri commented:
I also have some questions:

- Did any of the test projects break due to compile errors? I saw this with newer Eclipse releases where the test classpath is separate from regular classes.
- Does classgraph support PDE devMode classpaths now?

apeteri commented:
In the Travis build, Tycho did not like `JavaSE-11` as an execution environment value: https://travis-ci.org/b2ihealthcare/snow-owl/builds/469543966

Commit:
[upgrade] Remove merge conflicts and set VMType to Java-11

Commit:
Merge remote-tracking branch 'origin/7.x' into feature/java_11

Commit:
[copyright] Update copyright header date

Commit:
Merge remote-tracking branch 'origin/7.x' into feature/java_11

zoliMolnar commented:
@apeteri Neither of the test projects broke due to compilation errors. 
ClassGraph supports it but currently it has a bug with the newer versions of eclipse (specifically versions from 2018-09). To resolve this I've forked ClassGraph, fixed the bug and opened a PR for it.

cmark commented:
Please bump Tycho to v1.3 to be able to build projects with Java 11.
https://wiki.eclipse.org/Tycho/Release_Notes/1.3

Commit:
[releng] Bump tycho version to 1.3.0 classgraph version to 4.6.9

Commit:
[releng] Configure Java11 for the maven build launch config

Commit:
Merge remote-tracking branch 'origin/7.x' into feature/java_11

Commit:
[docs] Add Java11 support for lgtm

Commit:
[releng] Bump swagger to 2.9.2, spring to 5.1.3, hateoas to 0.25.0 ...

... spring-security to 5.1.2

Commit:
[releng] Bump classgraph to 4.6.10 jackson to 2.9.0 javax to 2.3.0

Commit:
[releng] Change jdk for tycho maven builds to jdk11

Commit:
[releng] Launch config for Java11

Commit:
Merge remote-tracking branch 'origin/7.x' into feature/java_11

Commit:
[upgrade] Change repository location to the 4.10 update site

Commit:
[upgrade] Bump guava to 21.0.0

Commit:
[upgrade] Fix guava and jackson version dependencies in manifests

Commit:
[releng] Fix mapdb project name in the project file

Commit:
[upgrade] Resolve conflicts caused by guava ugprade

Commit:
Merge remote-tracking branch 'origin/7.x' into feature/java_11

Commit:
[upgrade] Bump swagger model and annotaions version to 1.5.14

Commit:
[refactor] Remove test code from ClassPathScanner.class

Commit:
[releng] Add Jaxb dependencies to restassured package

Commit:
[releng] Remove cleanerHackEnable method call to fix tests

Commit:
[releng] Launch config modifications

Commit:
Merge remote-tracking branch 'origin/7.x' into feature/java_11

Commit:
[releng] Add javax.annotation.api bundle to rest manifests

Commit:
Merge remote-tracking branch 'origin/7.x' into feature/java_11

cmark committed:
Merge branch '7.x' into feature/java_11

Conflicts:
	commons/com.b2international.commons/.classpath
	core/com.b2international.snowowl.api.impl/.classpath
	core/com.b2international.snowowl.api.impl/META-INF/MANIFEST.MF
	core/com.b2international.snowowl.api.rest/.classpath
	core/com.b2international.snowowl.api.rest/META-INF/MANIFEST.MF
	core/com.b2international.snowowl.api.rest/build.properties
	core/com.b2international.snowowl.api.rest/pom.xml
	core/com.b2international.snowowl.api/.classpath
	core/com.b2international.snowowl.api/META-INF/MANIFEST.MF
	core/com.b2international.snowowl.api/src/com/b2international/snowowl/api/codesystem/ICodeSystemService.java
	core/com.b2international.snowowl.core/META-INF/MANIFEST.MF
	core/com.b2international.snowowl.dependencies/.classpath
	core/com.b2international.snowowl.fhir.api/.classpath
	core/com.b2international.snowowl.fhir.api/META-INF/MANIFEST.MF
	core/com.b2international.snowowl.fhir.api/build.properties
	core/com.b2international.snowowl.fhir.api/pom.xml
	pom.xml
	releng/target-platform/.launch/snow-owl-oss.launch
	snomed/com.b2international.snowowl.snomed.api.rest.tests/.launch/com.b2international.snowowl.snomed.api.rest.tests-win32_x86_64
(ext-tp).launch
	snomed/com.b2international.snowowl.snomed.api.rest/.classpath
	snomed/com.b2international.snowowl.snomed.api.rest/META-INF/MANIFEST.MF
	snomed/com.b2international.snowowl.snomed.api.rest/build.properties
	snomed/com.b2international.snowowl.snomed.api.rest/pom.xml
	snomed/com.b2international.snowowl.snomed.datastore/META-INF/MANIFEST.MF
	snomed/com.b2international.snowowl.snomed.datastore/src/com/b2international/snowowl/snomed/datastore/IdStorageKeyPair.java
	snomed/com.b2international.snowowl.snomed.datastore/src/com/b2international/snowowl/snomed/datastore/request/rf2/SnomedRf2ImportRequest.java
	snomed/com.b2international.snowowl.snomed.datastore/src/com/b2international/snowowl/snomed/datastore/taxonomy/InvalidRelationship.java
	tests/com.b2international.restassured-all/META-INF/MANIFEST.MF
	tests/com.b2international.snowowl.test.commons/META-INF/MANIFEST.MF

cmark committed:
[build] fix compile and runtime issues with Java 11

cmark committed:
[build] fix restassured bundle manifest

cmark committed:
Merge branch '7.x' into feature/java_11

cmark committed:
[build] fix build issues

Remove duplicate dependency entries.
Remove javax.xml from server.dependencies.feature.

cmark committed:
[rf2] remove unnecessary cleanerHackEnable call

cmark committed:
[tp] add Xtext 2.18 dependencies to target platform

Bump Guava to 21.0.
Bump Protege to 5.0.4-b2i.
Remove jackson datatype guava module.

cmark committed:
[build] bump Tycho to 1.4.0

Bump Xtend compiler to 2.18.

cmark committed:
[log] add logback.xml to build.properties

cmark committed:
[tp] bump Eclipse Platform to 4.11

cmark committed:
[build] create proper Equinox server package

Bump Jetty to v9.4.15.
Fix launch arguments in snow-owl-oss.product.
Add bundle start levels to snow-owl-oss.product.

cmark committed:
[logging] use configuration/serviceability.xml for Logback configuration

cmark committed:
[build] remove id and application props

cmark committed:
[build] remove Virgo packaging

cmark committed:
[build] add custom packaging of Equinox server

Add custom snowowl.bat and snowowl.sh startup scripts (WIP).

cmark committed:
[build] add Debian packaging (WIP)

cmark committed:
[http] add jetty XMLs to configuration/etc

Configure jetty.home and jetty log class during Snow Owl initialization.

cmark committed:
[http] move jetty.xmls to configuration directory

cmark committed:
[log] remove snow-owl directory from log.log path

cmark committed:
[build] update docs link

cmark committed:
[build] commit progress on new Debian and updated RPM packages

cmark committed:
[build] remove jetty sysprops from product launching section

cmark committed:
[build] configure jetty only when it is not configured via sysprop

cmark committed:
[releng] remove no longer valid license entries from NOTICE

Update license year.

cmark committed:
[build] fix startup script path in snowowl.service and rpm/init.d script

cmark committed:
[build] fix cygwin issue in snowowl.sh

cmark committed:
[launch] fix core-e2e-tests.launch

cmark committed:
[plugin] remove platform admin dependency from ClassPathScanner

cmark committed:
[tests] disable stack trace trimming

cmark committed:
[deps] bump typetools to 0.6.1

cmark committed:
[deps] bump classgraph to 4.8.43

cmark committed:
[plugins] slightly improve classgraph scan performance

cmark committed:
Revert "[plugins] slightly improve classgraph scan performance"

This reverts commit df303212dde2c96097e767c59a1fb26b4fec38d9.

cmark committed:
[tests] increase memory to 2gb in SNOMED CT API tests
